ui/app/AppLayouts/Chat/ChatColumn/Message.qml

```diff
@@ -318,6 +338,69 @@ Item {
             visible: false
         }
     }
+
+    Repeater {
+      model: imageUrls.split(" ")
+      visible: messageWrapper.appSettings.displayChatImages && imageUrls != ""
+      Rectangle {
+        property int chatVerticalPadding: 7
+        property int chatHorizontalPadding: 12
+
+        id: chatBox2
+        height: {
+          if (!messageWrapper.appSettings.displayChatImages || imageUrls == "") {
+            return 0
+          }
+          return (isCurrentUser || (!isCurrentUser && authorCurrentMsg == authorPrevMsg) ? childrenRect.height : 24 + childrenRect.height)
+        }
+        color: isCurrentUser ? Theme.blue : Theme.lightBlue
+        border.color: Theme.transparent
+        width:  350 + 2 * chatHorizontalPadding
+        radius: 16
+        anchors.left: !isCurrentUser ? chatImage.right : undefined
+        anchors.leftMargin: !isCurrentUser ? 8 : 0
+        anchors.right: !isCurrentUser ? undefined : parent.right
+        anchors.rightMargin: !isCurrentUser ? 0 : Theme.padding
+        anchors.top: (index == 0) ? chatBox.bottom : parent.children[index-1].bottom
+        anchors.topMargin: messageWrapper.appSettings.displayChatImages ? chatBox2.chatVerticalPadding: 0
+        visible: (isMessage || isEmoji) && messageWrapper.appSettings.displayChatImages && imageUrls != ""
+
+        // Thi`s rectangle's only job is to mask the corner to make it less rounded... yep
+        Rectangle {
```

Comment:
Do we reuse that fake rectangle often? Might be worth putting in Component (only for the chat still, no need to put it in shared)

Reply:
Absolutely. Same goes for the `chatBox` thing. Will fine-tune/refactor once the everything works well with rendering